dbms/src/Storages/DeltaMerge/Delta/MemTableSet.cpp

```diff
@@ -28,25 +28,6 @@ namespace DM
 {
 void MemTableSet::appendColumnFileInner(const ColumnFilePtr & column_file)
 {
-    // If this column file's schema is identical to last_schema, then use the last_schema instance (instead of the one in `column_file`),
-    // so that we don't have to serialize my_schema instance.
-    if (auto * m_file = column_file->tryToInMemoryFile(); m_file)
-    {
-        auto my_schema = m_file->getSchema();
-        if (last_schema && my_schema && last_schema != my_schema && isSameSchema(*my_schema, *last_schema))
-            m_file->resetIdenticalSchema(last_schema);
-        else
-            last_schema = my_schema;
-    }
-    else if (auto * t_file = column_file->tryToTinyFile(); t_file)
-    {
-        auto my_schema = t_file->getSchema();
-        if (last_schema && my_schema && last_schema != my_schema && isSameSchema(*my_schema, *last_schema))
-            t_file->resetIdenticalSchema(last_schema);
-        else
-            last_schema = my_schema;
-    }
```

Comment:
Please add a test (if there is no) verifying that identical schemas are not serialized repeatedly.

For example, if your "SharedBlockSchemas" returns different pointers (but their data is the same) to the same schema, I guess we will serialize repeatedly.